apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/AbstractTracingSpan.java

```diff
@@ -301,6 +302,12 @@ public void ref(TraceSegmentRef ref) {
         if (refs == null) {
             refs = new LinkedList<>();
         }
+        /*
+         * Provide the OOM protection if the entry span hosts too many references.
+         */
+        if (refs.size() >= Config.Agent.TRACE_SEGMENT_REF_LIMIT_PER_SPAN) {
```

Comment:
```suggestion
        if (refs.size() == Config.Agent.TRACE_SEGMENT_REF_LIMIT_PER_SPAN) {
```
What is the case should be `>`? Unless there is a bug here, or faces face condition, but actually race condition is a bug

Reply:
Fixed.